CoursesManager.UI/ViewModels/ConfigurationViewModel.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        private string _mailPort;
+        public string MailPort
+        {
+            get => _mailPort;
+            set => SetProperty(ref _mailPort, value);
+        }
+
+        private string _mailUser;
+        public string MailUser
+        {
+            get => _mailUser;
+            set => SetProperty(ref _mailUser, value);
+        }
+
+        private string _mailPassword;
+        public string MailPassword
+        {
+            get => _mailPassword;
+            set => SetProperty(ref _mailPassword, value);
+        }
+
+        
+
+        private EnvModel _appConfig;
+        public EnvModel AppConfig
+        {
+            get => _appConfig;
+            set => SetProperty(ref _appConfig, value);
+        }
+
+        public ICommand SaveCommand { get; }
+
+        public ConfigurationViewModel(IConfigurationService configurationService, string dbServer, string dbPort, string dbUser, string dbPassword, string dbName, string mailServer, string mailPort, string mailUser, string mailPassword, EnvModel appConfig)
+        {
+            _configurationService = configurationService;
+            _dbServer = dbServer;
+            _dbPort = dbPort;
+            _dbUser = dbUser;
+            _dbPassword = dbPassword;
+            _dbName = dbName;
+            _mailServer = mailServer;
+            _mailPort = mailPort;
+            _mailUser = mailUser;
+            _mailPassword = mailPassword;
+            _appConfig = appConfig;
+            InitializeSettings();
+            SaveCommand = new RelayCommand(ValidateAndSave, CanSave);
+        }
+
+        public void InitializeSettings()
+        {
+            var dbParams = _configurationService.GetDatabaseParameters();
+            DbServer = dbParams.TryGetValue("Server", out var server) ? server : string.Empty;
+            DbPort = dbParams.TryGetValue("Port", out var port) ? port : string.Empty;
+            DbUser = dbParams.TryGetValue("User", out var user) ? user : string.Empty;
+            DbPassword = dbParams.TryGetValue("Password", out var password) ? password : string.Empty;
+            DbName = dbParams.TryGetValue("Database", out var dbName) ? dbName : string.Empty;
+
+            var mailParams = _configurationService.GetMailParameters();
+            MailServer = mailParams.TryGetValue("Server", out var mailServer) ? mailServer : string.Empty;
+            MailPort = mailParams.TryGetValue("Port", out var mailPort) ? mailPort : string.Empty;
+            MailUser = mailParams.TryGetValue("User", out var mailUser) ? mailUser : string.Empty;
+            MailPassword = mailParams.TryGetValue("Password", out var mailPassword) ? mailPassword : string.Empty;
+
+            Console.WriteLine($"DbServer: {DbServer}, DbPort: {DbPort}, DbUser: {DbUser}, DbPassword: {DbPassword}, DbName: {DbName}");
+            Console.WriteLine($"MailServer: {MailServer}, MailPort: {MailPort}, MailUser: {MailUser}, MailPassword: {MailPassword}");
+        }
```

Comment:
Aangezien je in InitializeSettings toch de waardes uit de _configurationService haalt is het wat overbodig deze ook in de constructor mee te geven. Daarbij is wat je in de constructor krijgt voor dbServer en dbPort enzo toch altijd null in de huidige context van de app.

Reply:
heb het uit de constructor gehaald